core/src/main/scala/io/finch/Accept.scala

```diff
@@ -0,0 +1,22 @@
+package io.finch
+
+import cats.Eq
+import javax.activation.MimeType
+import scala.util.control.NonFatal
+
+case class Accept(primary: String, subtype: String)
+
+object Accept {
+
+  def apply(s: String): Option[Accept] = try {
+    val mt = new MimeType(s)
+    Some(Accept(mt.getPrimaryType, mt.getSubType))
+  } catch {
+    case NonFatal(_) => None
+  }
+
+  implicit val eq: Eq[Accept] = Eq.instance((a, b) => {
```

Comment:
* Symmetry and reflexivity don't hold. (*/*, text/html).
* If type is `*` then subtype should be `*` too. 
```
media-range    = ( "*/*"
                        | ( type "/" "*" )
                        | ( type "/" subtype )
                        ) *( ";" parameter )
```
from https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

Reply:
Hi

It's not supposed to be symmetric. We match client accept range against server range. While client could accept any wildcard media type, server can't produce wildcard one.

Though, I agree that using equality type class for it could be obscuring.

Reply:
Imho it worths efforts to introduce other type class in despite of only one possible instance. Violation of the law (especially such natural law) can become a time delayed bug.

Reply:
I think it was me who suggested using the `Eq` type class. Maybe it wasn't such a great idea (sorry for misleading suggestions). @ImLiar maybe we should/can roll back to `Accept.matches` (or something)?

Reply:
Let's do that as a follow-up PR.